tests/scripts/all.sh

```diff
@@ -34,13 +34,26 @@ MEMORY=0
 SHORT=0
 FORCE=0
 
+OPENSSL="openssl"
+OPENSSL_LEGACY="$OPENSSL"
```

Comment:
I disagree with this, when I introduced the possibility to change both options, I was thinking that someone just has installed an older version of OpenSSL and GnuTLS and would only like to test using those (which would be the default). In that case you would still like to run all tests but your `legacy` and newer version are effectively the same. Also, notice that adding the `--no-legacy-tests` is not straight forward because the tests in `compat.sh` and `ssl-opt.sh` do not have any version requirements and legacy and other tests are mixed in the same run of the scripts.